firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1844,63 +1844,120 @@ class BrowserViewController: UIViewController,
         }
     }
 
-    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
+    /// Returns whether the credit card autofill feature flag is enabled from Nimbus
+    func autofillCreditCardNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.creditCardAutofillStatus, checking: .buildOnly)
+    }
+
+    /// Returns whether credit card autofill settings are enabled based on user defaults key KeyAutofillCreditCardStatus
+    func autofillCreditCardSettingsUserDefaultIsEnabled() -> Bool {
         let userDefaults = UserDefaults.standard
         let keyCreditCardAutofill = PrefsKeys.KeyAutofillCreditCardStatus
 
-        let autofillCreditCardStatus = featureFlags.isFeatureEnabled(
-            .creditCardAutofillStatus, checking: .buildOnly)
+        return (userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true)
+    }
+
+    /// Returns whether the address autofill feature flag is enabled from Nimbus
+    func addressAutofillNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.addressAutofill, checking: .buildOnly)
+    }
+
+    /// Returns whether address autofill settings are enabled based on user defaults.
+    func addressAutofillSettingsUserDefaultsIsEnabled() -> Bool {
+        let userDefaults = UserDefaults.standard
+        let keyAddressAutofill = PrefsKeys.KeyAutofillAddressStatus
+
+        return (userDefaults.object(forKey: keyAddressAutofill) as? Bool ?? true)
+    }
+
+    /// Sets up credit card autofill handling.
+    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
```

Comment:
nit: change the name to `autofillFormSetup` because now it handles credit card and also addresses?

unless it doesn't 🤔 @jnrahme thoughts on this?

Reply:
it needs a name change